addons/product_extended/product_extended.py

```diff
@@ -41,43 +23,39 @@ def compute_price(self, cr, uid, product_ids, template_ids=False, recursive=Fals
             ids = template_ids
             model = 'product.template'
         for prod_id in ids:
-            bom_obj = self.pool.get('mrp.bom')
+            MrpBom = self.env['mrp.bom']
             if model == 'product.product':
-                bom_id = bom_obj._bom_find(cr, uid, product_id=prod_id, context=context)
+                bom_id = MrpBom._bom_find(product_id=prod_id)
             else:
-                bom_id = bom_obj._bom_find(cr, uid, product_tmpl_id=prod_id, context=context)
+                bom_id = MrpBom._bom_find(product_tmpl_id=prod_id)
             if bom_id:
                 # In recursive mode, it will first compute the prices of child boms
+                bom = MrpBom.browse(bom_id)
                 if recursive:
-                    #Search the products that are components of this bom of prod_id
-                    bom = bom_obj.browse(cr, uid, bom_id, context=context)
-
                     #Call compute_price on these subproducts
                     prod_set = set([x.product_id.id for x in bom.bom_line_ids])
```

Comment:
I think this could be `bom.bom_line_ids.mapped('product_id.id')`, to verify /cc @rco-odoo